### DISCUSSION TITLE
metal noise in Melgan, HiFiGan, VITS

### DISCUSSION THREAD

**Discussion by dmazurok** [coqui-ai/TTS#1814] [General Q&A]
Hi!
I'm trying to train a good TTS, but can't get GAN vocoder to work well enough. 
First I tried to train MelGan, FB-Melgan, HiFiGan, VITS models on real data - metal-like noise persists. Not everywhere, just in some specific sounds. 
Then I tried to use Tacotron outputs (with force alignment) and train aforementioned vocoders on them - didn't help much (or didn't help at all).
I trained WaveGrad model - it works perfect... but the way too slow.
I tried to generate extra data with WaveGrad and Tacotron and use it to train MelGan and HiFiGan - no success.
On spectrogram, there are some small differences in specific sounds, that causes GAN-based vocoders to produce that noise. But I haven't encountered such a problem with any other voices. Seems like that particular voice due to its features can't be used with GAN-based vocoders...  I hope I'm mistaken. 
Please, if there are any ideas worth to try, share with me!
Example is attached below. Thanks a lot!
[hifigan_Sent1_1.zip](https://github.com/coqui-ai/TTS/files/9237053/hifigan_Sent1_1.zip)

**Comment by domcross** [comment]
The GAN-style vocoders tend to produce metallic artifacts. After some hundred thousand trainings the artifcats should slowly go away. How many steps did you train your vocoder-models?

**Reply by dmazurok** [reply]
I had many tries, from 600k steps to 2M steps, still that sound persists. It seems like it's always overfitting, I used 30 hours of speech with different voices - no success. I used 30k of this single speaker - no success.

**Reply by senseiGURU** [reply]
@dmazurok any luck with this? I'm having the exact same issue. Were you able to find a solution?

**Reply by MaxIakovliev** [reply]
I'm having same  issue.
I'm training single voice. Around 2 hours of more or less clean voice.
Still having metal noise. 
So far I tried to:
1. normalize
2. de-noise every audio file
Any update/ideas how to fix it?

**Comment by Bilallbayrakdar** [comment]
This happened to me when I used different audio configs in TTS and Vocoder models. You should check the configs if you didn't.
